src/vm/database/sqlite.rs

```diff
@@ -154,3 +154,15 @@ impl SqliteConnection {
         &mut self.conn
     }
 }
+
+
+#[cfg(test)]
+#[test]
+#[should_panic]
+fn test_rollback() {
+    let mut conn = SqliteConnection::memory().unwrap();
+    let bhh = BlockHeaderHash([1; 32]);
+    conn.begin(&bhh);
+    conn.rollback(&bhh);
+    conn.commit(&bhh); // shouldn't be on the stack!
```

Comment:
Should we write this test to make clear that the panic should happen on `commit`?

Reply:
Sure -- 70b62da54 makes this a more specific test.

Reply:
Thank you @kantai!